conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+    HINT_SOURCE = "hint-source"
+    HINT_GRAYSKULL = "hint-grayskull"
+    UPDATE_ALL = "update-all"
+    UPDATE_SOURCE = "update-source"
+    UPDATE_GRAYSKULL = "update-grayskull"
+
+
+class BotConfig(BaseModel):
+    """This dictates the behavior of the conda-forge auto-tick bot which issues automatic version updates/migrations for feedstocks."""
+
+    automerge: Annotated[
+        Union[bool, BotConfigAutoMergeChoice],
+        Field(description="Automatically merge PRs if possible"),
+    ] = False
+
+    check_solvable: Annotated[
+        Union[bool, None],
+        Field(
+            description="Open PRs only if resulting environment is solvable."
+        ),
+    ] = None
+
+    inspection: Annotated[
+        Union[BotConfigInspectionChoice, None],
+        Field(
+            description="Method for generating hints or updating recipe",
+        ),
+    ] = None
+
+    abi_migration_branches: Annotated[
+        Union[List[str], None],
+        Field(description="List of branches for additional bot migration PRs"),
+    ] = None
+
+    version_updates_random_fraction_to_keep: Annotated[
+        Union[float, None],
+        Field(
+            description="Fraction of versions to keep for frequently updated packages"
+        ),
+    ] = None
+
+
+class ChannelPriorityConfig(str, Enum):
+    STRICT = "strict"
+    FLEXIBLE = "flexible"
+    DISABLED = "disabled"
+
+
+class CondaForgeChannels(BaseModel):
+    """This represents the channels to grab packages from during builds and which channels/labels to push to on anaconda.org after a package has been built. The channels variable is a mapping with sources and targets."""
+
+    sources: List[str] = Field(
+        default=["conda-forge"],
+        description="sources selects the channels to pull packages from, in order",
+    )
+
+    targets: List[List[str]] = Field(
+        default=[["conda-forge", "main"]],
+        description="targets is a list of 2-lists, where the first element is the channel to push to and the second element is the label on that channel",
+    )
+
+
+class CondaBuildConfig(BaseModel):
+    pkg_format: Union[int, str, None] = Field(
+        description="The package format for conda build. This can be one of 1, 2, or tar. The default is 2.",
+        default="2",
+    )
+
+    zstd_compression_level: int = Field(
+        default=16,
+        description="The compression level for the zstd compression algorithm for .conda artifacts. conda-forge uses a default value of 16 since its artifacts can be large.",
+    )
+
+    @field_validator("pkg_format")
+    def pkg_format_validator(cls, v):
+        if not v:
+            return v
+
+        valid_values = ["2", None]
+        if v not in valid_values:
+            raise ValueError(
+                f"Invalid value {v} for pkg_format. Valid values are {valid_values}"
+            )
+        return v
+
+
+class CondaForgeDocker(BaseModel):
+    model_config = ConfigDict(extra="allow")
```

Comment:
Do we want to allow extra keys? Why?

Reply:
removed and moved all fields to optional values, in case one of the supported config values is passed validation will occur, in case of extra variables in the config, right now the behavior is to ignore it (will not show up when dumping to dict/json)